### PR TITLE
Refactor Sharding-transaction.

### PR DESCRIPTION
**Pull request by cherrylzhao**

### CODE REVIEW

File: sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/connection/ShardingConnectionTest.java [hunk 1]
```diff
@@ -93,8 +89,8 @@ public void clear() {
         try {
             connection.close();
             TransactionTypeHolder.clear();
-            FixedXAShardingTransactionHandler.getInvokes().clear();
-            FixedBaseShardingTransactionHandler.getInvokes().clear();
+//            FixedXAShardingTransactionHandler.getInvokes().clear();
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/connection/ShardingConnectionTest.java]: Why comment here, should remove comment codes

File: sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/datasource/ShardingDataSourceTest.java [hunk 2]
```diff
@@ -178,7 +177,7 @@ public void assertGetXaConnectionThenGetLocalConnection() throws SQLException {
         ShardingConnection shardingConnection = shardingDataSource.getConnection();
         assertThat(shardingConnection.getDataSourceMap().size(), is(1));
         assertThat(shardingConnection.getTransactionType(), is(TransactionType.XA));
-        assertThat(shardingConnection.getShardingTransactionHandler(), instanceOf(FixedXAShardingTransactionHandler.class));
+//        assertThat(shardingConnection.getShardingTransactionHandler(), instanceOf(FixedXAShardingTransactionHandler.class));
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/datasource/ShardingDataSourceTest.java]: Why comment here, should remove comment codes

File: sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/connection/ShardingConnectionTest.java [hunk 3]
```diff
@@ -120,21 +116,21 @@ public void assertRelease() throws SQLException {
...
         connection = new ShardingConnection(dataSourceMap, shardingContext, TransactionType.XA);
         connection.setAutoCommit(false);
-        assertThat(FixedXAShardingTransactionHandler.getInvokes().get("begin"), instanceOf(ShardingTransactionEvent.class));
+//        assertThat(FixedXAShardingTransactionHandler.getInvokes().get("begin"), instanceOf(ShardingTransactionContext.class));
         connection.commit();
-        assertThat(FixedXAShardingTransactionHandler.getInvokes().get("commit"), instanceOf(ShardingTransactionEvent.class));
+//        assertThat(FixedXAShardingTransactionHandler.getInvokes().get("commit"), instanceOf(ShardingTransactionContext.class));
         connection.rollback();
-        assertThat(FixedXAShardingTransactionHandler.getInvokes().get("rollback"), instanceOf(ShardingTransactionEvent.class));
+//        assertThat(FixedXAShardingTransactionHandler.getInvokes().get("rollback"), instanceOf(ShardingTransactionContext.class));
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/connection/ShardingConnectionTest.java]: Why comment here, should remove comment codes

File: sharding-proxy/src/test/java/io/shardingsphere/shardingproxy/backend/jdbc/connection/BackendTransactionManagerTest.java [hunk 4]
```diff
@@ -39,10 +37,10 @@
     
     private BackendTransactionManager backendTransactionManager = new BackendTransactionManager(backendConnection);
     
-    @BeforeClass
-    public static void beforeClass() {
-        ShardingTransactionHandlerRegistry.load();
-    }
+//    @BeforeClass
+//    public static void beforeClass() {
+//        ShardingTransactionHandlerRegistry.load();
```
Comment (terrymanu) [inline sharding-proxy/src/test/java/io/shardingsphere/shardingproxy/backend/jdbc/connection/BackendTransactionManagerTest.java]: Why comment here, should remove comment codes

File: sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/datasource/ShardingDataSourceTest.java (hunk 2 above)
Comment (cherrylzhao) [reply sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/datasource/ShardingDataSourceTest.java]: I have fixed it ,comment out later.

File: sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/connection/ShardingConnectionTest.java (hunk 3 above)
Comment (cherrylzhao) [reply sharding-jdbc/sharding-jdbc-core/src/test/java/io/shardingsphere/shardingjdbc/jdbc/core/connection/ShardingConnectionTest.java]: I have fixed it ,comment out later.

File: sharding-proxy/src/test/java/io/shardingsphere/shardingproxy/backend/jdbc/connection/BackendTransactionManagerTest.java (hunk 4 above)
Comment (cherrylzhao) [reply sharding-proxy/src/test/java/io/shardingsphere/shardingproxy/backend/jdbc/connection/BackendTransactionManagerTest.java]: yes ,this can be removed.

**Pull request merged**